sdk/eventhub/Azure.Messaging.EventHubs/design/partition-receiver-proposal.md

```diff
@@ -0,0 +1,451 @@
+
+These developers are interested in using the low-level components of the Event Hubs client library, focused around client-service communication that they can control and customize to meet their needs. This class of developers are considered advanced users of Event Hubs with a deep understanding of the service, cloud development, and messaging systems. Many are willing to accept the complexity of working with lower-level components for the ability to have more control for their implementation.
+
+## Why this is needed
+
+Because the `EventHubConsumerClient` is not bound to a specific partition, it cannot directly interact with a partition to read events.  In order to read events from a partition, the client needs to spawn a transport consumer to communicate with the service by creating an AMQP link and manage the state for reading from the partition, such as the identifier of partition and the position wthin the event stream being read.
+
+While it would be possible to add a `ReceiveBatch` method directly on the consumer client, that introduces challenges for providing a clean and understandable API and for managing resource use.  The theoretical `ReceiveBatch` method would have to have a signature similar to `ReceiveBatch(string partition, EventPosition position);`  Calling the method would create an AMQP link if one did not exist for the partition, set the position, and read a batch of events.  
+
+The challenges begin after the first call.  Should a developer invoke the method more than once, they would have to advance the `EventPosition` argument to match the last event read to express the intent of "continue where we left off," otherwise, the existing AMQP link for that partition would have to be closed and a new one created at the desired position.  Ultimately, this places a heavy burden on callers, having to track a virtual checkpoint after each `ReceiveBatch` call or would result in the undesirable behavior of network resources being created/destroyed often.   Should the link not be destroyed within the scope of the call, a measure of non-determinism is introduced in which the developer lacks the ability to express the intent to stop reading from that partition and clean up resources.
+
+For more context and examples, please refer to the [addendum](#addendum-why-this-is-needed-by-example).
+
+## High level scenarios
+
+### A multiplayer game can issue rewards in real-time
+
+A popular multiplayer game powers its experience through an ecosystem of services, each responsible for different aspects of gameplay and player experience.  As part of their operation, the sevices communicate with each other as well as integrate with other downstream systems.  Azure Event Hubs is used as a means of passive coordination to avoid direct calls withing the ecosystem.
```

Comment:
```suggestion
A popular multiplayer game powers its experience through an ecosystem of services, each responsible for different aspects of gameplay and player experience.  As part of their operation, the services communicate with each other as well as integrate with other downstream systems.  Azure Event Hubs is used as a means of passive coordination to avoid direct calls withing the ecosystem.
```

Reply:
Thanks!   For some reason, spell checking stopped working and had to be toggled; got this and a couple of other mistakes.

Reply:
Sure :)